File: 240510/정수 명령 처리 2/process-numeric-commands-2.java
import java.util.*;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Queue<Integer> queue = new LinkedList<Integer>();

        for (int i = 0; i < n; i++) {
            String cmd = sc.next();
            if(cmd.equals("push")){
                int item = sc.nextInt();
                queue.add(item);
            }else if(cmd.equals("pop")){
                System.out.println(queue.poll());
            }else if(cmd.equals("size")) {
                System.out.println(queue.size());
            }else if (cmd.equals("empty")) {
                if(queue.isEmpty()) {
                    System.out.println(1);
                }else{
                    System.out.println(0);
                }
            }else if (cmd.equals("front")){
                System.out.println(queue.peek());
            }
        }

    }
}

File: 240510/정수 명령 처리 2/process-numeric-commands-2.kt
import java.util.*



fun main(){

    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val queue: Queue<Int> = LinkedList<Int>()

    for(i in 1 .. n){
        val cmd = sc.next()
        when(cmd) {
            "push" -> {
                val item = sc.nextInt()
                queue.add(item)
            }
            "front" -> {
                println(queue.peek())
            }
            "size" -> {
                println(queue.size)
            }
            "empty" -> {
                if(queue.isEmpty()) {
                    println(1)
                }else {
                    println(0)
                }
            }
            "pop" -> {
                println(queue.poll())
            }
        }
    }
    
}